src/RpcServer/RpcServer.cs

```diff
@@ -47,6 +58,14 @@ public RpcServer(NeoSystem system, RpcServerSettings settings)
             localNode = system.LocalNode.Ask<LocalNode>(new LocalNode.GetInstance()).Result;
             RegisterMethods(this);
             Initialize_SmartContract();
+
+            ApplicationEngine.Log += Ev;
+            Blockchain.Committed += OnCommitted;
+        }
+
+        private void OnCommitted(NeoSystem system, Block block)
+        {
+            LogEvents.Clear();
```

Comment:
Should we be clearing all the events or just the ones for the transactions in `block`?

Reply:
I think we should clear all log events, otherwise some logs will be there forever, which will be a memory leak.

Reply:
If you clear them in `OnCommitting`, other plugins that depend on this data may not work. I think it should be cleaned up in `OnCommitted` by a certain plugin.